homeassistant/components/litterrobot/update.py

```diff
@@ -0,0 +1,107 @@
+"""Support for Litter-Robot updates."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from datetime import datetime, timedelta
+from typing import Any
+
+from pylitterbot import LitterRobot4
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.start import async_at_start
+
+from .const import DOMAIN
+from .entity import LitterRobotEntity, LitterRobotHub
+
+FIRMWARE_UPDATE_ENTITY = UpdateEntityDescription(
+    key="firmware",
+    name="Firmware",
+    device_class=UpdateDeviceClass.FIRMWARE,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Litter-Robot update platform."""
+    hub: LitterRobotHub = hass.data[DOMAIN][entry.entry_id]
+    robots = hub.account.robots
+    entities = [
+        RobotUpdateEntity(robot=robot, hub=hub, description=FIRMWARE_UPDATE_ENTITY)
+        for robot in robots
+        if isinstance(robot, LitterRobot4)
+    ]
+    async_add_entities(entities)
+
+
+class RobotUpdateEntity(LitterRobotEntity[LitterRobot4], UpdateEntity):
+    """A class that describes robot update entities."""
+
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL | UpdateEntityFeature.PROGRESS
+    )
+
+    def __init__(
+        self,
+        robot: LitterRobot4,
+        hub: LitterRobotHub,
+        description: UpdateEntityDescription,
+    ) -> None:
+        """Initialize a Litter-Robot update entity."""
+        super().__init__(robot, hub, description)
+        self._poll_unsub: Callable[[], None] | None = None
+        self._attr_installed_version = robot.firmware
```

Comment:
Doesn't look like this will update in the update function below

Reply:
Thanks for the catch